src/WebJobs.Script.WebHost/Scale/TableStorageScaleMetricsRepository.cs

```diff
@@ -54,40 +58,49 @@ public TableStorageScaleMetricsRepository(IConfiguration configuration, IHostIdP
             _delegatingHandlerProvider = delegatingHandlerProvider ?? throw new ArgumentNullException(nameof(delegatingHandlerProvider));
         }
 
-        internal CloudTableClient TableClient
+        internal TableServiceClient TableServiceClient
         {
             get
             {
-                if (_tableClient == null)
+                if (_tableServiceClient is null)
                 {
                     string storageConnectionString = _configuration.GetWebJobsConnectionString(ConnectionStringNames.Storage);
-                    if (!string.IsNullOrEmpty(storageConnectionString) &&
-                        CloudStorageAccount.TryParse(storageConnectionString, out CloudStorageAccount account))
+
+                    try
                     {
-                        var restConfig = new RestExecutorConfiguration { DelegatingHandler = _delegatingHandlerProvider.Create() };
-                        var tableClientConfig = new TableClientConfiguration { RestExecutorConfiguration = restConfig };
+                        DelegatingHandler handler = _delegatingHandlerProvider.Create();
+                        TableClientOptions options = null;
 
-                        _tableClient = new CloudTableClient(account.TableStorageUri, account.Credentials, tableClientConfig);
+                        if (handler != null)
+                        {
+                            options = new TableClientOptions
+                            {
+                                Transport = new HttpClientTransport(handler)
+                            };
+                        }
+
+                        _tableServiceClient = new TableServiceClient(storageConnectionString, options);
                     }
-                    else
+                    catch (Exception ex)
```

Comment:
Whenever catching plain `Exception` type, recommend checking it is not fatal:

```suggestion
                    catch (Exception ex) when (!ex.IsFatal())
```